### PR TITLE
chore(examples): added Gatsby hot reload example

### PR DESCRIPTION
**Pull request by thsig**
This example project showcases Gatsby running inside a container, with
support for hot reloading.

### CODE REVIEW

File: examples/gatsby-hot-reload/minimal/garden.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  hotReload:
+    sync:
+      - target: /app/
+  services:
+    - name: minimal
+      command: [npm, run, dev]
+      hotReloadCommand: [npm, run, dev]
+      env:
+        # GATSBY_WEBPACK_PUBLICPATH: /
+        GATSBY_WEBPACK_PUBLICPATH: http://gatsby-hot-reload.local.app.garden/
```
Comment (drubin) [inline examples/gatsby-hot-reload/minimal/garden.yml]: Can we try `GATSBY_WEBPACK_PUBLICPATH: /` with the latest version of Gatsby? At least `gatsby@2.0.42`

Also mind linking to the public issue https://github.com/gatsbyjs/gatsby/issues/8348#issuecomment-435418042

File: examples/gatsby-hot-reload/minimal/README.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
+<p align="center">
```
Comment (drubin) [inline examples/gatsby-hot-reload/minimal/README.md]: Can you please make this rather linked to garden usage? 

Similar to these 

https://github.com/garden-io/garden/blob/master/examples/remote-k8s/README.md 
https://github.com/garden-io/garden/tree/master/examples/local-tls/README.md

**Comment by drubin** [comment]
@thsig mind giving this a 2nd check? 

I fixed all the issues i pointed now and improved the example to only mount `src`

File: examples/gatsby-hot-reload/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+# Example of Gatsby using hot reload
+
+This project shows how you can configure gatsby and garden local development on Kubernetes.
+
+## Usage
+
+Develop on gatsby:
+
+In this example we use the standard `dev` command but we tell garden to hot reload the files from `src` into the container with out rebuilding
```
Comment (eysi09) [inline examples/gatsby-hot-reload/README.md]: _without_

File: examples/gatsby-hot-reload/gatsby-config.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+module.exports = {
+  siteMetadata: {
+    title: 'Gatsby Default Starter',
+  },
+  plugins: [
+    'gatsby-plugin-react-helmet',
+    // {
```
Comment (eysi09) [inline examples/gatsby-hot-reload/gatsby-config.js]: Can't we remove this?

File: examples/gatsby-hot-reload/gatsby-ssr.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+/**
```
Comment (eysi09) [inline examples/gatsby-hot-reload/gatsby-ssr.js]: Delete this file as well?

File: examples/gatsby-hot-reload/package.json [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    "react-dom": "^16.5.1",
+    "react-helmet": "^5.2.0"
+  },
+  "keywords": [
+    "gatsby"
+  ],
+  "license": "MIT",
+  "scripts": {
+    "build": "gatsby build",
+    "dev": "gatsby develop -H 0.0.0.0 -p 8000",
```
Comment (eysi09) [inline examples/gatsby-hot-reload/package.json]: Perhaps we should add a note about the `dev` command to the readme.

File: examples/gatsby-hot-reload/package.json [hunk 7]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "gatsby-hot-reload",
+  "description": "Gatsby hot reload demo",
+  "version": "1.0.0",
+  "author": "Kyle Mathews <mathews.kyle@gmail.com>",
```
Comment (eysi09) [inline examples/gatsby-hot-reload/package.json]: Set `"garden.io <info@garden.io>"` as author.

File: examples/gatsby-hot-reload/package.json (hunk 6 above)
Comment (drubin) [reply examples/gatsby-hot-reload/package.json]: How do you imagine this working? The point is that this is a garden example so explaining how to run it without garden felt odd to me.

I included the `garden dev --hot-reload=minimal` in the readme

Comment (eysi09) [reply examples/gatsby-hot-reload/package.json]: Whoops, I meant why we use host flag, `-H 0.0.0.0`. Didn't make that clear in my first comment.

Comment (drubin) [reply examples/gatsby-hot-reload/package.json]: Ah yes sure can do.

File: examples/gatsby-hot-reload/gatsby-config.js (hunk 4 above)
Comment (drubin) [reply examples/gatsby-hot-reload/gatsby-config.js]: @eysi09  I am not really sure how Gatsby works (and whats required) do you want to take a stab and removing all the unnecessary bits?

**Pull request merged**